config/action.d/iptables-allports.conf

```diff
@@ -21,6 +23,10 @@ actionstart = <iptables> -N f2b-<name>
               <iptables> -A f2b-<name> -j <returntype>
               <iptables> -I <chain> -p <protocol> -j f2b-<name>
 
+actionstart6 = <ip6tables> -N f2b-<name>6
+               <ip6tables> -A f2b-<name>6 -j <returntype>
+               <ip6tables> -I <chain> -p <protocol> -j f2b-<name>6
```

Comment:
Understood. I will omit it where ´possible.
However I agree that it's desireable having a general way of handling this.
But for such a cross-plattform software like fail2ban, it's heavily dependent on the plattform specifics being used to enforce the blocking. E.g.

1. Linux iptables needs dedicated start/stop/ban and unban action commands
2. Linux iptables with ipset currently as well but set of type inet may really become protocol independent so that only the start/stop commands would be come protocol dependent.
3. For iptables, not only the binary name differs but also protocol dependent things like the blocktype
4. hostsdeny is completely protocol independent (no dedicated ban/unban commands)
5. route as well
6. PF on *BSD is also completely protocol independent and works like iptables with ipset
7. ...

So your approach is very challenging and I'm not sure if we can reach this right away. I personally would concentrate on introducing v6 support at all.

What happens at the moment when a real v6 address is captured from the logs: the ban/unban commands just return with a unsuccessful return code.

So any thing we do in this matter is a improvement and may not be perfect and leave room for further improvements. For the KISS principal, I see such a enhancement as such a future improvement. Another approach to achieve this could also be a script that offloads this logic from fail2ban instead of calling the actual commands directly.